### DISCUSSION TITLE
Getting headers informations on static pages ?

### DISCUSSION THREAD

**Discussion by Romzeer** [vercel/next.js#21300] [Help]
Hi,

I have a little question, I need to access to the header of request for getting user agent of my user and i'm wondering if there is a way to do it with static pages? 
I know that I can access to it if I use SSR but I want to do it in the static way. 
Does I need to wait for window object loaded with an event listener ?
Any recommandations ? 

Thanks in advance

**Comment by toyamarinyon** [comment]
How about using the `useEffect` hook? This will only be executed during the rendering phase, so the window object will always to be present.
Eventlister is also fine, but I think it is more "React" way.


```
export default function SomePage() {
  const [ua, setUa] = useState("");
  useEffect(() => {
    console.log(window.navigator.userAgent);
    setUa(window.navigator.userAgent)
  }, []);

  return <div>{ua}</div>
}
```
